operator/src/main/java/org/keycloak/operator/controllers/KeycloakIngress.java

```diff
@@ -97,8 +97,8 @@ private Ingress newIngress() {
                 .endSpec()
                 .build();
 
-        if (!keycloak.getSpec().isHostnameDisabled()) {
-            ingress.getSpec().getRules().get(0).setHost(keycloak.getSpec().getHostname());
+        if (keycloak.getSpec().getHostnameSpec() != null && keycloak.getSpec().getHostnameSpec().getHostname() != null) {
```

Comment:
I also agree that setting the `hostname` is enough most of the time.

But we also have the `hostname-url` option to force a base URL to frontend URLs. That said, I'm not sure if you want to resolve the host from the `hostname-url` too in case this option is set.

Reply:
Good point. 

I'd say we could parse the hostname from the `hostname-url`, if the `hostname` is not specified. Otherwise, prefer only the `hostname`. 

If both are specified, it's the responsibility of the user/dev to manage them, right?!